plumbing/msg/sender.go

```diff
@@ -56,22 +55,38 @@ func (s *Sender) Send(ctx context.Context, from, to address.Address, value *type
 	s.l.Lock()
 	defer s.l.Unlock()
 
-	nonce, err := nextNonce(ctx, s.chainReader, s.msgPool, from)
+	st, err := s.chainReader.LatestState(ctx)
+	if err != nil {
+		return cid.Undef, err
```

Comment:
could be wrapped for more clarity, eg "couldnt get latest state: " here and below